### PR TITLE
[CRD] Warning if multiple Dyankubes with extensions are enabled for same ApiURL

### PR DESCRIPTION
**Pull request by andriisoldatenko**
## Description

Ticket [DAQ-1617](https://dt-rnd.atlassian.net/browse/DAQ-1617)

Currently having multiple Dynakubes with extensions enabled for the same tenant in the same K8S cluster will have unwanted side-effects. OTOH we don’t want to completely prevent this scenario.

This PR introduce warning validation check for this scenario.

## How can this be tested?

unit tests/ or create 2 dks extensions enabled with same APIURLs and see no-blocking warning:
```
You are already using a Dynakube that enables extensions. Having multiple Dynakubes with '.spec.extensions' enabled can have severe side-effects on “sum” and “count” metrics and cause double-billing.
```

[DAQ-1617]: https://dt-rnd.atlassian.net/browse/DAQ-1617?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by StefanHauth — APPROVED** [review]
LGTM

File: pkg/api/validation/dynakube/eec.go [hunk 1]
```diff
@@ -24,6 +26,31 @@ func extensionControllerImage(_ context.Context, _ *Validator, dk *dynakube.Dyna
 	return ""
 }
 
+func warnIfmultiplyDKwithExtensionsEnabled(ctx context.Context, dv *Validator, dk *dynakube.DynaKube) string {
```
Comment (0sewa0) [inline pkg/api/validation/dynakube/eec.go:29]:
```suggestion
func conflictingTenantForExtensions(ctx context.Context, dv *Validator, dk *dynakube.DynaKube) string {
```

File: pkg/api/validation/dynakube/eec.go [hunk 2]
```diff
@@ -3,11 +3,13 @@ package validation
 import (
 	"github.com/Dynatrace/dynatrace-operator/pkg/api/v1beta3/dynakube"
 	"golang.org/x/net/context"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 const (
 	errorExtensionExecutionControllerImageNotSpecified       = `DynaKube's specification enables the Prometheus feature, make sure you correctly specify the ExtensionExecutionController image.`
 	errorExtensionExecutionControllerInvalidPVCConfiguration = `DynaKube specifies a PVC for the extension controller while ephemeral volume is also enabled. These settings are mutually exclusive, please choose only one.`
+	warningIfmultiplyDKwithExtensionsEnabled                 = `You are already using a Dynakube that enables extensions. Having multiple Dynakubes with '.spec.extensions' enabled can have severe side-effects on “sum” and “count” metrics and cause double-billing.`
```
Comment (0sewa0) [inline pkg/api/validation/dynakube/eec.go:12]:
```suggestion
	warningConflictingTenantForExtensions                 = `You are already using a Dynakube that enables extensions. Having multiple Dynakubes with '.spec.extensions' enabled can have severe side-effects on “sum” and “count” metrics and cause double-billing.`
```

File: pkg/api/validation/dynakube/eec.go (hunk 1 above)
Comment (0sewa0) [reply pkg/api/validation/dynakube/eec.go:29]: you could also call it `conflictingApiUrlForExtensions`, to be more precious

File: pkg/api/validation/dynakube/eec.go (hunk 2 above)
Comment (0sewa0) [reply pkg/api/validation/dynakube/eec.go:12]: I like the message, but it doesn't mention that the problems is that both DynaKube use the same tenant (`apiUrl`), should we mention that for clarity? (if we mention it, then I would stick to calling out the `apiUrl` being the same, as maybe a customer will be not familiar with the "tenant" naming)

File: pkg/api/validation/dynakube/eec.go (hunk 1 above)
Comment (0sewa0) [reply pkg/api/validation/dynakube/eec.go:29]: [(based it on already existing validation func names)](https://github.com/Dynatrace/dynatrace-operator/blob/f85df8f8f7fb174496f39f32bc94cba84c8656cd/pkg/api/validation/dynakube/validation.go#L37-L42)

**Review by 0sewa0 — COMMENTED** [review]
Logic looks good, just some naming things

File: pkg/api/validation/dynakube/eec.go (hunk 2 above)
Comment (andriisoldatenko) [reply pkg/api/validation/dynakube/eec.go:12]: @StefanHauth wdyt?

File: pkg/api/validation/dynakube/eec.go (hunk 1 above)
Comment (andriisoldatenko) [reply pkg/api/validation/dynakube/eec.go:29]: on it!

File: pkg/api/validation/dynakube/eec.go (hunk 2 above)
Comment (andriisoldatenko) [reply pkg/api/validation/dynakube/eec.go]: renamed in bab9b2f5e

Comment (StefanHauth) [reply pkg/api/validation/dynakube/eec.go]: Good catch, that should definitely be in the message. Also the name of the already existing DK should be in there.

```
`You are already using a Dynakube ('<name of existing DK>') that enables extensions. Having multiple Dynakubes with same '.spec.apiUrl' and '.spec.extensions' enabled can have severe side-effects on “sum” and “count” metrics and cause double-billing.`
```

Comment (andriisoldatenko) [reply pkg/api/validation/dynakube/eec.go]: fixed in 40c4c4e5d531e55fe236236dad3b487658bffcfd

Comment (andriisoldatenko) [reply pkg/api/validation/dynakube/eec.go]: @0sewa0, @StefanHauth  PTAL!

**Review by 0sewa0 — APPROVED** [review]

**Comment by andriisoldatenko** [comment]
tested in my cluster:
```
kaf ~/work/samples/dk-eec.yaml
Warning: ActiveGate specification missing memory limits. Can cause excess memory usage.
Warning: You are already using a Dynakube ('dynakube') that enables extensions. Having multiple Dynakubes with same '.spec.apiUrl' and '.spec.extensions' enabled can have severe side-effects on “sum” and “count” metrics and cause double-billing.
dynakube.dynatrace.com/dynakube1 created
```

**Pull request merged**